Commands/Principals/SetGroupPermissions.cs

```diff
@@ -1,75 +1,102 @@
-                    {
-                        roleDefinitionBindings.Remove(roleDefinition);
-                        roleAssignment.Update();
-                        ClientContext.ExecuteQueryRetry();
-                        break;
-                    }
-                }
-            }
-        }
-    }
-}
+﻿using System.Linq;
+using System.Management.Automation;
+using Microsoft.SharePoint.Client;
+using OfficeDevPnP.PowerShell.CmdletHelpAttributes;
+using OfficeDevPnP.PowerShell.Commands.Base.PipeBinds;
+
+namespace OfficeDevPnP.PowerShell.Commands.Principals
+{
+    [Cmdlet(VerbsCommon.Set, "SPOGroupPermissions")]
+    [CmdletHelp("Adds and/or removes permissions of a specific SharePoint group",
+        Category = CmdletHelpCategory.Principals)]
+    [CmdletExample(
+        Code = @"PS:> Set-SPOGroupPermissions -Identity 'My Site Members' -AddRole Contribute",
+        Remarks = "Adds the 'Contribute' permission to the SharePoint group with the name 'My Site Members'",
+        SortOrder = 1)]
+    [CmdletExample(
+        Code = @"PS:> Set-SPOGroupPermissions -Identity 'My Site Members' -RemoveRole 'Full Control' -AddRole 'Read'",
+        Remarks = "Removes the 'Full Control' from and adds the 'Contribute' permissions to the SharePoint group with the name 'My Site Members'",
+        SortOrder = 2)]
+    [CmdletExample(
+        Code = @"PS:> Set-SPOGroupPermissions -Identity 'My Site Members' -AddRole @('Contribute', 'Design')",
+        Remarks = "Adds the 'Contribute' and 'Design' permissions to the SharePoint group with the name 'My Site Members'",
+        SortOrder = 3)]
+    [CmdletExample(
+        Code = @"PS:> Set-SPOGroupPermissions -Identity 'My Site Members' -RemoveRole @('Contribute', 'Design')",
+        Remarks = "Removes the 'Contribute' and 'Design' permissions from the SharePoint group with the name 'My Site Members'",
+        SortOrder = 4)]
+    public class SetGroupPermissions : SPOWebCmdlet
+    {
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = "ByName", HelpMessage = "Get the permissions of a specific group by name")]
+        [Alias("Name")]
+        public GroupPipeBind Identity = new GroupPipeBind();
+
+        [Parameter(Mandatory = false, HelpMessage = "The list to apply the command to.")]
+        public ListPipeBind List = new ListPipeBind();
+
+        [Parameter(Mandatory = false, HelpMessage = "Name of the permission set to add to this SharePoint group")]
+        public string[] AddRole = null;
+
+        [Parameter(Mandatory = false, HelpMessage = "Name of the permission set to remove from this SharePoint group")]
+        public string[] RemoveRole = null;
+
+        protected override void ExecuteCmdlet()
+        {
+            var group = Identity.GetGroup(SelectedWeb);
+
+            List list = null;
+            if (List != null)

```

Comment:
Given that you initialize the List parameter in line 35 as a new ListPipeBind(), it will never be null...